src/core/server/http/http_service.ts

```diff
@@ -69,7 +73,8 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
 
     const config = await this.config$.pipe(first()).toPromise();
 
-    return this.httpServer.setup(config);
+    const setup = this.httpServer.setup(config);
+    return { ...setup, ...{ createNewServer: (cfg: HttpConfig) => this.createServer(cfg) } };
```

Comment:
Can you elaborate on why `createNewServer` should not exist as a method on `httpServer.setup(config)`?

Reply:
`createNewServer` needs to return a new instance of the `HttpServer` class itself. I've never really liked "factory" patterns like this:

```js
class Test {
  createNewTest () {
    return new Test()
  }
}
```

I can move it over to HttpService if that's where we think it better belongs though